plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -625,6 +634,58 @@ def finalize(self):
                 add_metadata_file_metadata('updateinfo', self.context.metadata_file_path,
                                            self.context.checksum)
 
+    def _get_pkglist_to_publish(self, erratum_unit):
+        """
+        Make a list of packages which will be listed in the published erratum.
+
+        It is possible to filter packages only if `repo_unit_nevra` is available,
+        otherwise publish everything except duplicated collections.
+
+        :param erratum_unit: the erratum unit to analyze
+        :param type: pulp_rpm.plugins.db.models.Errata
+        :return: list of pkglists to publish
+        :rtype: list of dicts
+        """
+        pkglists_to_publish = []
+
+        # If we can pull a repo_id off the conduit, use that to generate repo-specific nevra
+        if self.context.conduit and hasattr(self.context.conduit, 'repo_id'):
+            repo_unit_nevra = self.context.get_repo_unit_nevra(erratum_unit)
+        else:
+            repo_unit_nevra = None
```

Comment:
I put this condition inside the loop for one reason only - not to publish duplicated pkglists.
Pkglist can be duplicated because of the merge logic (assume that all the repos contain the erratum with the same id):
 - if you create multiple repos with the same feed to sync from and then sync, for each repo pkglist will be added
 - if you create repo by copy from another one, no changes will be made to erratum pkglists.

I think it is worth the trouble because errata pkglists look more sane then. What do you think?